lib/Runtime/Library/JavascriptBigIntObject.cpp

```diff
@@ -0,0 +1,76 @@
+//-------------------------------------------------------------------------------------------------------
+// Copyright (C) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE.txt file in the project root for full license information.
+//-------------------------------------------------------------------------------------------------------
+#include "RuntimeLibraryPch.h"
+
+namespace Js
+{
+    JavascriptBigIntObject::JavascriptBigIntObject(JavascriptBigInt* value, DynamicType * type)
+        : DynamicObject(type), value(value)
+    {
+        Assert(type->GetTypeId() == TypeIds_BigIntObject);
+    }
+
+    bool JavascriptBigIntObject::Is(Var aValue)
+    {
+        return JavascriptOperators::GetTypeId(aValue) == TypeIds_BigIntObject;
+    }
+
+    JavascriptBigIntObject* JavascriptBigIntObject::FromVar(Js::Var aValue)
+    {
+        AssertOrFailFastMsg(Is(aValue), "Ensure var is actually a 'JavascriptBigIntObject'");
+
+        return static_cast<JavascriptBigIntObject *>(aValue);
+    }
+
+    JavascriptBigIntObject* JavascriptBigIntObject::UnsafeFromVar(Js::Var aValue)
+    {
+        AssertMsg(Is(aValue), "Ensure var is actually a 'JavascriptBigIntObject'");
+
+        return static_cast<JavascriptBigIntObject *>(aValue);
+    }
+
+    JavascriptBigInt* JavascriptBigIntObject::GetValue() const
+    {
+        if (this->value == nullptr)
```

Comment:
I don't understand why this->value would ever be null, but JavascriptBooleanObject and JavascriptSymbolObject both include this check. JavascriptNumberObject does not allow null. Anyhow, it caught my attention because `false` is a sort of surprising thing to return. Maybe just skip this check entirely, so it returns nullptr if the value is nullptr?

Reply:
Okay, I did some code archaeology and found that the uninitialized state (and the Initialize method below) are used in subclassing built-in types, because getting the correct `this` during a subclass constructor is a bit complicated.

---
In reply to: [218154672](https://github.com/Microsoft/ChakraCore/pull/5706#discussion_r218154672) [](ancestors = 218154672)

Reply:
Never mind, it was originally for subclassing but the implementation has changed a lot. Now it seems only used for time-travel debugging, in the method CreateDefaultBoxedObject_TTD.

---
In reply to: [218163468](https://github.com/Microsoft/ChakraCore/pull/5706#discussion_r218163468) [](ancestors = 218163468,218154672)